Lib/test/test_pprint.py

```diff
@@ -323,6 +323,18 @@ def test_width(self):
      '1 '
      '2']]]]]""")
 
+    def test_integer(self):
+        self.assertEqual(pprint.pformat(1234567), '1234567')
+        self.assertEqual(pprint.pformat(1234567, underscore_numbers=True), '1_234_567')
```

Comment:
Can you add an additional:

`self.assertEqual(pprint.pformat((True, False), underscore_numbers=True), '(True, False)')`

we want to ensure that bool still gets its repr rather than 0 or 1.

Reply:
Yes, I can add it. However, I wonder if it's better to add:
```python
self.assertEqual(pprint.pformat(simple, underscore_numbers=True), native)
```
in `def test_same_as_repr()` [before this assertion](https://github.com/python/cpython/blob/94faa0724f8cbae6867c491c8e465e35f4fdbfbb/Lib/test/test_pprint.py#L209). So it checks more types (including boolean values).

What do you think about this solution? I will add the preferred one.

Reply:
good idea, do that instead. :)

Reply:
I committed it.

I have made the requested changes; please review again